hare/flows_test.go

```diff
@@ -217,21 +217,20 @@ func (m *mockClock) LayerToTime(layer types.LayerID) time.Time {
 	return m.layerTime[layer]
 }
 
-func (m *mockClock) AwaitLayer(layer types.LayerID) chan struct{} {
+func (m *mockClock) AwaitLayer(ctx context.Context, layer types.LayerID) context.Context {
 	m.m.Lock()
 	defer m.m.Unlock()
 
+	ctx, cancel := context.WithCancel(ctx)
+
 	if _, ok := m.layerTime[layer]; ok {
-		ch := make(chan struct{})
-		close(ch)
-		return ch
-	}
-	if ch, ok := m.channels[layer]; ok {
-		return ch
+		cancel()
+		return ctx
 	}
-	ch := make(chan struct{})
-	m.channels[layer] = ch
-	return ch
+	cancels := m.layersSubscriptions[layer]
+	cancels = append(cancels, cancel)
+	m.layersSubscriptions[layer] = cancels
```

Comment:
I think this can probably be shortened to:

```suggestion
	m.layersSubscriptions[layer] = append(m.layersSubscriptions[layer], cancel)
```